### PR TITLE
feat(eks): Support KubernetesVersion 1.18

### PR DESCRIPTION
**Pull request by pahud**
Support `KubernetesVersion` 1.18

Fixed: #10853

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-eks/README.md [hunk 1]
```diff
@@ -270,7 +270,7 @@ The following code defines an Amazon EKS cluster with a default Fargate Profile
 
 ```ts
 const cluster = new eks.FargateCluster(this, 'MyCluster', {
-  version: eks.KubernetesVersion.V1_16,
+  version: eks.KubernetesVersion.V1_18,
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-eks/README.md:273]: seems like there are a bunch of examples within the readme. (`KubernetesVersion.V1_16` is referenced 4x and `KubernetesVersion.V1_17` is referenced 7x)

is there a specific reason to only update 2 of them?

**Review by shivlaks — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-eks/README.md (hunk 1 above)
Comment (iliapolo) [reply packages/@aws-cdk/aws-eks/README.md:273]: Good point. Might as well update all of them.

Comment (pahud) [reply packages/@aws-cdk/aws-eks/README.md:273]: Really appreciated! Let me do it!

**Comment by pahud** [comment]
will update the tests shortly

**Comment by pahud** [comment]
all fixed!

**Review by iliapolo — APPROVED** [review]

**Pull request merged**